sdl-core/src/main/scala/io/smartdatalake/workflow/ActionDAGRunState.scala

```diff
@@ -72,10 +90,11 @@ private[smartdatalake] object ActionDAGRunState {
     {case obj: DataObjectId => obj.id}
   ))
 
-  implicit private lazy val workflowReflections = ReflectionUtil.getReflections("io.smartdatalake.workflow")
+  implicit private lazy val workflowReflections: Reflections = ReflectionUtil.getReflections("io.smartdatalake.workflow")
 
   private lazy val typeHints = ShortTypeHints(ReflectionUtil.getTraitImplClasses[SubFeed].toList ++ ReflectionUtil.getSealedTraitImplClasses[ExecutionId], "type")
-  implicit private val formats: Formats = Json4sCompat.getStrictSerializationFormat(typeHints) + new EnumNameSerializer(RuntimeEventState) +
+  implicit val formats: Formats = Serialization.formats(typeHints)
+    .withStrictArrayExtraction.withStrictMapExtraction.withStrictOptionParsing + new EnumNameSerializer(RuntimeEventState) +
```

Comment:
This seems to be a wrong merge conflict resultion. We need to use Json4sCompat.getStrictSerializationFormat to be compatible with older json4s versions of Spark 3.1.
```suggestion
    Json4sCompat.getStrictSerializationFormat(typeHints) + new EnumNameSerializer(RuntimeEventState) +
```